android/rctmgl/src/main/java/com/mapbox/rctmgl/components/styles/RCTMGLStyle.java

```diff
@@ -57,16 +59,24 @@ public RCTMGLStyleValue getStyleValueForKey(String styleKey) {
     }
 
     public void addImage(String uriStr) {
-        if (uriStr == null || isTokenString(uriStr)) {
+        if (!shouldAddImage(uriStr)) {
             return;
         }
-
         Map.Entry[] images = new Map.Entry[]{ new AbstractMap.SimpleEntry(uriStr, uriStr) };
         DownloadMapImageTask task = new DownloadMapImageTask(mMap, null);
         task.execute(images);
     }
 
+    private boolean shouldAddImage(String uriStr) {
+        return uriStr != null && !isTokenString(uriStr) && isValidURI(uriStr);
```

Comment:
Do we still need `!isTokenString` when we already have `isValidURI`?

Reply:
Good catch, we do not I'll update it.